### ISSUE TITLE
defaultProps not works for optional props in strict null checking mode

### ISSUE THREAD

**Issue by clinyong** [Microsoft/TypeScript#16964]
**TypeScript Version:**  2.4.1

**Code**

Hello.tsx

```ts
import * as React from "react";

export interface HelloProps {
	list?: number[];
}

export class Hello extends React.Component<HelloProps, {}> {
	static defaultProps = {
		list: [1, 2, 3]
	};

	render() {
		const { list } = this.props;
		return (
			<ul>
				{// error here: Object is possibly 'undefined'.
				list.map(item =>
					<li>
						{item}
					</li>
				)}
			</ul>
		);
	}
}
```

tsconfig.json

```json
{
    "compilerOptions": {
        "outDir": "./dist/",
        "sourceMap": true,
        "noImplicitAny": true,
        "module": "commonjs",
        "target": "es5",
        "jsx": "react",
        "strictNullChecks": true
    },
    "include": [
        "src/**/*"
    ]
}
```

**Expected behavior:**

Notice that I set `defaultProps` here. So I think it should not report error.

**Actual behavior:**

```
ERROR in ./src/Hello.tsx
(19,21): error TS2532: Object is possibly 'undefined'.
```

**Comment by nortonwong** [comment]
This issue isn't with TypeScript, but with the React type definitions. There is actually [already an issue for this there](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/11640).
